cpprevolve/revolve/gazebo/brains/DifferentialCPG.cpp

```diff
@@ -279,17 +285,24 @@ DifferentialCPG::DifferentialCPG(
     }
   }
 
+
+
   // Create directory for output.
-  this->directory_name = controller->GetAttribute("output_directory")->GetAsString();
+//  this->directory_name = controller->GetAttribute("output_directory")->GetAsString();
   if(this->directory_name.empty())
   {
-    this->directory_name = "output/cpg_bo/";
-    this->directory_name += std::to_string(time(0)) + "/";
+      std::cout << "§yes§";
+    this->directory_name = "output/cpg_bo/" + this->robot->GetName() + "/"; // CHANGETHIS
+    this->directory_name += this->battery_->time_init + "/";
   }
+  else
+      std::cout << "§no§\n";
+
+
 
-  std::system(("mkdir -p " + this->directory_name).c_str());
+    std::system(("mkdir -p " + this->directory_name).c_str());
```

Comment:
There has to be a better way to this (this is only UNIX for example). Also it could be a security issue